cmd/podman/cp.go

```diff
@@ -350,3 +367,60 @@ func streamFileToStdout(srcPath string, srcfi os.FileInfo) error {
 	}
 	return nil
 }
+
+func isVolumeDestName(path string, ctr *libpod.Container) (bool, string, string) {
+	separator := string(os.PathSeparator)
+	if filepath.IsAbs(path) {
+		path = strings.TrimPrefix(path, separator)
+	}
+	if path == "" {
+		return false, "", ""
+	}
+	path = strings.Split(path, separator)[0]
+	for _, vol := range ctr.Config().NamedVolumes {
+		volNamePath := strings.TrimPrefix(vol.Dest, separator)
+		volBaseName := strings.Split(volNamePath, separator)[0]
+		if volBaseName == path {
+			return true, vol.Dest, vol.Name
+		}
+	}
+	return false, "", ""
+}
+
+// if SRCPATH or DESTPATH is from volume mount's destination, generate the path with volume mount point
+func pathWithVolumeMount(ctr *libpod.Container, runtime *libpod.Runtime, volDestName, volName, path string) (string, error) {
+	destVolume, err := runtime.GetVolume(volName)
+	if err != nil {
+		return "", errors.Wrapf(err, "error getting volume destination %s", volName)
+	}
+	if !filepath.IsAbs(path) {
+		path = filepath.Join(string(os.PathSeparator), path)
+	}
+	path = filepath.Join(destVolume.MountPoint(), strings.TrimPrefix(path, volDestName))
+	return path, nil
+}
+
+func isBindMountDestName(path string, ctr *libpod.Container) (bool, specs.Mount) {
+	separator := string(os.PathSeparator)
+	if filepath.IsAbs(path) {
+		path = strings.TrimPrefix(path, string(os.PathSeparator))
+	}
+	if path == "" {
+		return false, specs.Mount{}
+	}
+	path = strings.Split(path, separator)[0]
```

Comment:
Wait, how does this work? We split the original path on separators, then we compare it against the entire path of the volume (minus leading slash)? This doesn't seem like it makes sense.

Say I have two volumes, `/tmp/test/a` and `/tmp/test/b`, and I want to copy into `/tmp/test/b` - what's to prevent this from matching `/tmp/test/a` here?

Reply:
also check the prefix? `if mDest == path && strings.HasPrefix(destpath, m.Destination)`
I'll also search the libraries.

Reply:
@mheon, changed the path match logic `func matchVolumePath(path, target string) bool`, PTAL